tests/chainer_tests/functions_tests/normalization_tests/test_standardize.py

```diff
@@ -0,0 +1,106 @@
+import functools
+import unittest
+
+import numpy
+
+from chainer.functions.normalization._standardize import _standardize
+from chainer import testing
+from chainer import utils
+
+
+def _skip_if(cond, reason):
+    """Skip test if cond(self) is True"""
+    def decorator(impl):
+        @functools.wraps(impl)
+        def wrapper(self, *args, **kwargs):
+            if cond(self):
+                raise unittest.SkipTest(reason)
+            else:
+                impl(self, *args, **kwargs)
+        return wrapper
+    return decorator
+
+
+@testing.parameterize(*testing.product([
+    [
+        {'ch_out': 1},
+        {'ch_out': 5},
+    ],
+    [
+        {'size': 10},
+        {'size': 20},
+    ],
+    [
+        {'dtype': numpy.float64},
+        {'dtype': numpy.float32},
+        {'dtype': numpy.float16},
+    ],
+    [
+        # same (str): flag whether input elems are same values.
+        #   'no'   : all elems are randamly-chosen,
+        #   'equal': all elems are equal,
+        #   'near' : all elems are (randomly-chosen small values
+        #            + same value).
+        {'eps': 1e-5, 'same': 'no'},
+        {'eps': 1e-1, 'same': 'no'},
+        {'eps': 1e-1, 'same': 'equal'},
+        {'eps': 1e-1, 'same': 'near'},
+    ],
+]))
+@testing.backend.inject_backend_tests(
+    None,
+    # CPU tests
+    testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+    })
+    # ChainerX tests
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+)
+class TestStandardize(testing.FunctionTestCase):
+
+    # TODO(hitsgub): Delete this line once double backward of
+    # :func:`~chainer.functions._standardize` is implemented.
+    skip_double_backward_test = True
+
+    def setUp(self):
+        self.check_forward_options.update({'atol': 5e-3, 'rtol': 1e-2})
+        self.check_backward_options.update({'atol': 5e-3, 'rtol': 1e-2})
+        self.check_double_backward_options.update({'atol': 5e-3, 'rtol': 1e-2})
+
+    def generate_inputs(self):
+        shape = self.ch_out, self.size
+        if self.same in ('equal', 'near'):
+            # Make self.x have same values
+            x = numpy.ones(shape, self.dtype)
+            x *= numpy.random.uniform(-1, 1)
+            if self.same == 'near':
+                # Make self.x have slightly different values
+                zero_scale = 10. ** numpy.random.randint(-40, -3)
```

Comment:
The scale is too small to add to `x`.

Reply:
Changed the way to make an array that has small variance (according to the number of significant digits of float64).